packages/expo-ota/README.md

```diff
@@ -0,0 +1,155 @@
+# expo-ota
+
+Provides Over The Air updates feature for your react-native app, highly customizable with out of the box support for Expo infrastructure based updates.
+
+Supports Android and iOS platform.
+
+# Installation
+
+Make sure you have unimodules installed as described in [here](https://github.com/unimodules/react-native-unimodules/blob/master/README.md)
+
+If you do, just type add `expo-ota` to you npm dependencies.
+
+```bash
+#Either with npm
+npm add expo-ota
+
+#Or yarn
+yarn add expo-ota
+```
+
+# Basic configuration
+
+OTA module requires some basic configuration in your app's native code. Please locate your `MainApplication.java` or `MainApplication.kt` for Android application, and `AppDelegate.m` for iOS.
+
+## Android
+Make sure following code is present in your `MainApplication.java`
+
+```java
+private ExpoOTA expoOTA;
+
+    private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
+        ...
+        @Nullable
+        @Override
+        protected String getJSBundleFile() {
+            return expoOTA.getBundlePath();
+        }
+        ...
+    }
+
+ @Override
+    public void onCreate() {
+        super.onCreate();
+        SoLoader.init(this, /* native exopackage */ false);
+        expoOTA = ExpoOTA.init(this, BuildConfig.DEBUG);
+    }
+
+```
+
+## iOS
+
+Make sure following code is present in your `AppDelegate.m`
+
+```objc
+
+@implementation AppDelegate {
+  EXOta *ota;
+}
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
+{
+  ota = [EXOta new]; // This must be firs instruction within this method
+  ...
+}
+
+- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge {
+#ifdef DEBUG
+    return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
+#else
+  NSString *persistedBundle = [ota bundlePath];
+  if(persistedBundle != nil)
+  {
+    return [NSURL fileURLWithPath:persistedBundle];
+  } else
+  {
+    return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
+  }
+#endif
+}
+```
+
+## app.json
+
+You also need to create `app.json` file in root folder of your react-native application (alongside root `package.json`). If you've ejected from Expo, file should already be there. Example file looks like following:
+
+```json
+{
+  "name": "OTA",
+  "displayName": "expo-ota-template",
+  "expo": {
+    "name": "OTA Example",
+    "slug": "expo-template-ota",
+    "sdkVersion": "1.0.0",
```

Comment:
we need to keep the meaning of `sdkVersion` the same as in the managed workflow, because if we overload it to mean the binary version then people can't [use the expo client with their bare app](https://docs.expo.io/versions/v35.0.0/bare/using-expo-client/).

Reply:
Honestly, I was unaware that we support such scenario. But if we do, we need to tweak our logic accordingly.

I will update README and app.json schema with new field, `nativeVersion`, which will be required in bare apps (I do not know how to enforce that, yet).
I will compare downloaded and embedded manifests against both `sdkVersion` and `nativeVersion`, given they're both present.
We should add support for native version header in our updates API, which would work similar to sdkVersion. Let us discuss how we would approach that.

Reply:
After discussion with Stanisław we suggest, that we keep it as it is. W leave current meaning of `sdkVersion` and do not add any new field for keeping track of user's native code version whatsoever.

If users will want to distinguish which apps uses which bundles, they can use release channels for now, and we'll try to document that best we can.

Reply:
sgtm